From 7496e112ecfbf8ab622b46df1bc67c8e01661f81 Mon Sep 17 00:00:00 2001
From: Hans-Kristian Arntzen <post@arntzen-software.no>
Date: Thu, 5 Jan 2023 15:23:39 +0100
Subject: [PATCH] vkd3d-shader: Do not match root constant without checking
 visibility.

No idea why this ended up without checking it, since the DXBC path did
actually check this after all.

Signed-off-by: Hans-Kristian Arntzen <post@arntzen-software.no>
---
 libs/vkd3d-shader/dxil.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libs/vkd3d-shader/dxil.c b/libs/vkd3d-shader/dxil.c
index 9dfb6419..b11b2ad7 100644
--- a/libs/vkd3d-shader/dxil.c
+++ b/libs/vkd3d-shader/dxil.c
@@ -467,13 +467,13 @@ static dxil_spv_bool dxil_cbv_remap(void *userdata, const dxil_spv_d3d_binding *
 
     shader_interface_info = remap->shader_interface_info;
 
-    /* Try to map to root constant -> push constant.
-     * Do not consider shader visibility here, as DXBC path does not appear to do it either. */
+    /* Try to map to root constant -> push constant. */
     for (i = 0; i < shader_interface_info->push_constant_buffer_count; i++)
     {
         const struct vkd3d_shader_push_constant_buffer *push = &shader_interface_info->push_constant_buffers[i];
         if (push->register_space == d3d_binding->register_space &&
-            push->register_index == d3d_binding->register_index)
+            push->register_index == d3d_binding->register_index &&
+            dxil_match_shader_visibility(push->shader_visibility, d3d_binding->stage))
         {
             memset(vk_binding, 0, sizeof(*vk_binding));
             vk_binding->push_constant = DXIL_SPV_TRUE;
-- 
2.43.0

